kubernetes/cache/testing.go

```diff
@@ -48,3 +50,26 @@ func NewTestingCacheWithClients(t *testing.T, clients map[string]kubernetes.Clie
 	cf.SetClients(clients)
 	return newTestingCache(t, cf, conf)
 }
+
+// ConstantWaitForSync waits continuously with zero interval. Only use this for testing.
+func ConstantWaitForSync(stopCh <-chan struct{}, cacheSyncs ...cache.InformerSynced) bool {
+	for {
+		select {
+		case <-stopCh:
+			return false
+		default:
+			if allCachesSynced(cacheSyncs...) {
+				return true
+			}
+		}
+	}
```

Comment:
Rather than simply spinning (using CPU as fast as it can) would this help provide better efficiency?
```suggestion
	ticker := time.NewTicker(100 * time.Millisecond) // use this to give the loop a breather
	defer ticker.Stop()

	for {
		select {
		case <-stopCh:
			return false
		case <-ticker.C:
			if allCachesSynced(cacheSyncs...) {
				return true
			}
		}
	}
```

Reply:
No this is exactly what causes the slowdown in the tests now. This is how the `client-go` func is implemented that we normally use when not testing:
https://github.com/kubernetes/client-go/blob/c106b23895edc59ff05c65770a17e6a6d3caee66/tools/cache/shared_informer.go#L302-L303

https://github.com/kubernetes/client-go/blob/c106b23895edc59ff05c65770a17e6a6d3caee66/tools/cache/shared_informer.go#L327-L343

Essentially it adds 100ms to most of the tests which quickly adds up. Maybe if the ticker is really low like `1ms` then it would add a negligible amount of time to the tests.

Reply:
ah! ok.